src/components/contacts-page/contacts-page.component.js

```diff
@@ -1,114 +1,53 @@
 import React from 'react';
-import './ContactsPage.css';
-import ContactsList from './contacts-list';
-// import data from '../data/data';
-import AddContactForm from './AddContactForm';
-import Popup from './Popup';
-import {
-  validateInput,
-  generateId,
-  isContactUnique
-} from '../helpers/helper';
+import PropTypes from 'prop-types';
+import './contacts-page.style.css';
+import ContactsList from '../contacts-list';
+import AddContactForm from '../add-contact-form';
+import Popup from '../Popup';
 import {
   TITLE,
-  REGEX,
   WARNING_MESSAGES,
   APPROVE_FLAGS
-} from '../constants/constants';
+} from '../../constants/constants';
 
 class ContactsPage extends React.Component {
 
   constructor(props) {
     super(props);
     this.state = {
-      contacts: [],
       popoup: {
         showPopup: false,
         message: '',
-        contactIdToDelete: null,
         popupType: null
       }
     };
   }
-  
-  addContact = (firstName, lastName, image) => {
-    const { contacts } = this.state;
-    if (!(validateInput(firstName, REGEX.NAME)
-      && validateInput(lastName, REGEX.NAME)
-      && validateInput(image, REGEX.URL))) {
-      return;
-    }
-    const newContact = {
-      id: generateId(),
-      firstName,
-      lastName,
-      image,
-      likes: 0,
-      isChecked: false
-    };
-    if (!isContactUnique(contacts, newContact)) {
-      return;
-    }
-    this.setState({ contacts: [newContact, ...contacts] });
-  }
 
   areContactsWithLikes = () => {
-    const { contacts } = this.state;
+    const { contacts } = this.props;
     return contacts.some(item => item.likes);
   };
 
   clearAllContactsLikes = () => {
```

Comment:
Now this function only calls another function. Remove this and use the one from props directly where you need it